agent/engine/execcmd/manager_start_linux.go

```diff
@@ -135,35 +141,43 @@ func (m *manager) doStartAgent(ctx context.Context, client dockerapi.DockerClien
 	}
 	execRes, err := client.CreateContainerExec(ctx, containerId, execCfg, dockerclient.ContainerExecCreateTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [create]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 
 	seelog.Debugf("Task engine [%s]: created ExecCommandAgent for container: %s -> docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	err = client.StartContainerExec(ctx, execRes.ID, dockerclient.ContainerExecStartTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [pre-start]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: sent ExecCommandAgent start signal for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	inspect, err := client.InspectContainerExec(ctx, execRes.ID, dockerclient.ContainerExecInspectTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [docker inspect]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: inspect ExecCommandAgent for container: %s -> pid: %d, exitCode: %d, running:%v, err:%v",
 		task.Arn, containerId, inspect.Pid, inspect.ExitCode, inspect.Running, err)
 
 	if !inspect.Running { //TODO: [ecs-exec] retry only for certain exit codes?
-		return StartError{
+		execMD.StoppedReason = fmt.Sprintf("Execute Command Agent process exited with exit code: %d", inspect.ExitCode)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{
 			error:     fmt.Errorf("ExecCommandAgent could not start for container (exitCode=%d)", inspect.ExitCode),
 			retryable: true,
 		}
 	}
 	seelog.Infof("Task engine [%s]: started ExecCommandAgent for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
-	container.SetExecCommandAgentMetadata(&apicontainer.ExecCommandAgentMetadata{
-		PID:          strconv.Itoa(inspect.Pid),
-		DockerExecID: execRes.ID,
-		CMD:          execAgentCmd,
-	})
-	return nil
+	execMD.PID = strconv.Itoa(inspect.Pid)
+	execMD.DockerExecID = execRes.ID
+	execMD.CMD = execAgentCmd
+	execMD.StartedAt = time.Now()
+	execMD.Status = apicontainerstatus.ManagedAgentRunning
+	return execMD, nil
```

Comment:
shouldn't we be emitting a container event for successful agent start here?

Reply:
It's emitted by the caller(s) of this function in agent/engine/docker_task_engine.go (see comment above)

Reply:
my bad, I was thinking about emitting container event when we start exec agent for the first time:
https://github.com/aws/amazon-ecs-agent/blob/feature/ecs_exec/agent/engine/docker_task_engine.go#L1267-L1272 

are we planning to add this in a different PR?

Reply:
Added it.  (This was a pretty major miss on my part.)